File: 682/main.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Println(calPoints([]string{"5", "2", "C", "D", "+"}))  // 30
	fmt.Println(calPoints([]string{"5","-2","4","C","D","9","+","+"}))   // 27

}

func calPoints(operations []string) int {
	ans := 0
	l := []int{}
	for _, s := range operations {
		if s == "D" {
			l = append(l, l[len(l)-1] * 2)
		} else if s == "C" {
			// f, _ := strconv.Atoi(ops[i-1])
			l = l[:len(l)-1]
		} else if s == "+" {
			l = append(l, l[len(l)-2] + l[len(l)-1])
		} else {
			v, _ := strconv.Atoi(s)
			l = append(l, v)
		}
	}
	for _, v := range l {
		ans += v
	}
	return ans
}

File: 682/main.rs
struct Solution {}

impl Solution {
    pub fn cal_points(operations: Vec<String>) -> i32 {
        let mut ans = 0;
        let mut l = Vec::new();
        for s in operations {

            if s == "D" {
                l.push(l[l.len()-1] * 2);
            } else if s == "C" {
                l.remove(l.len()-1);
            } else if s == "+" {
                l.push(l[l.len()-2] + l[l.len() - 1]);
            } else {
                let n = s.parse::<i32>().unwrap();
                l.push(n);
            }
        }

        for i in l {
            ans += i;
        }
        return ans as i32;
    }
}

fn main() {
    println!("{}", Solution::cal_points(vec!["5".to_string(),"-2".to_string(),"4".to_string(),"C".to_string(),"D".to_string(),"9".to_string(),"+".to_string(),"+".to_string()]))
}